apps/web/src/pages/integrations/components/multi-provider/SelectPrimaryIntegrationModal.tsx

```diff
@@ -0,0 +1,261 @@
+  flex: 1 0 0;
+  margin-top: 14px;
+`;
+
+const columns: IExtendedColumn<ITableIntegration>[] = [
+  {
+    accessor: 'name',
+    Header: 'Name',
+    Cell: IntegrationNameCell,
+  },
+  {
+    accessor: 'provider',
+    Header: 'Provider',
+    Cell: withCellLoading(
+      ({ row: { original } }) => {
+        return (
+          <Text data-test-id="integration-provider-cell" rows={1}>
+            {original.provider}
+          </Text>
+        );
+      },
+      { loadingTestId: 'integration-provider-cell-loading' }
+    ),
+  },
+  {
+    accessor: 'active',
+    Header: 'Status',
+    width: 125,
+    maxWidth: 125,
+    Cell: IntegrationStatusCell,
+  },
+];
+
+export interface ISelectPrimaryIntegrationModalProps {
+  isOpened: boolean;
+  environmentId?: string;
+  channelType?: ChannelTypeEnum;
+  onClose: () => void;
+}
+
+export const SelectPrimaryIntegrationModal = ({
+  isOpened,
+  environmentId,
+  channelType,
+  onClose,
+}: ISelectPrimaryIntegrationModalProps) => {
+  const theme = useMantineTheme();
+  const [isPopoverOpened, setPopoverOpened] = useState(false);
+  const [{ selectedIntegrationId, isActive, selectedRowId }, setSelectedState] = useState<{
+    selectedIntegrationId?: string;
+    isActive: boolean;
+    selectedRowId?: string;
+  }>({ isActive: true });
+
+  const { environments, isLoading: areEnvironmentsLoading } = useFetchEnvironments();
+  const environmentName = environments?.find((el) => el._id === environmentId)?.name ?? '';
+
+  const { integrations, loading: areIntegrationsLoading } = useIntegrations();
+  const { makePrimaryIntegration, isLoading: isMarkingPrimaryIntegration } = useMakePrimaryIntegration({
+    onSuccess: onClose,
+  });
+  const integrationsByEnvAndChannel = useMemo<ITableIntegration[]>(() => {
+    const filteredIntegrations = (integrations ?? []).filter((el) => {
+      if (environmentId) {
+        return el.channel === channelType && el._environmentId === environmentId;
+      }
+
+      return el.channel === channelType;
+    });
+
+    return filteredIntegrations.map((el) => mapToTableIntegration(el, environments));
+  }, [integrations, environments, channelType, environmentId]);
+
+  const initialSelectedIndex = useMemo(() => {
+    const primary = findPrimaryIntegration(integrationsByEnvAndChannel ?? []);
+
+    return integrationsByEnvAndChannel?.indexOf(primary);
+  }, [integrationsByEnvAndChannel]);
+
+  const isLoading = areEnvironmentsLoading || areIntegrationsLoading;
+  const isInitialProviderSelected = !selectedRowId || selectedRowId === `${initialSelectedIndex}`;
+  const makePrimaryButtonDisabled = !selectedIntegrationId || isLoading || isInitialProviderSelected;
+  const channelName = CHANNEL_TYPE_TO_STRING[channelType ?? ''];
+
+  const onRowSelectCallback = (row: Row<ITableIntegration>) => {
+    setSelectedState({
+      selectedIntegrationId: row.original.integrationId,
+      isActive: row.original.active,
+      selectedRowId: row.id,
+    });
+  };
```

Comment:
when the integration is selected by clicking on radio button we do store the info in the state